libutils/proc_manager.h

```diff
@@ -0,0 +1,123 @@
+/*
+  Copyright 2021 Northern.tech AS
+
+  This file is part of CFEngine 3 - written and maintained by Northern.tech AS.
+
+  This program is free software; you can redistribute it and/or modify it
+  under the terms of the GNU General Public License as published by the
+  Free Software Foundation; version 3.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
+
+  To the extent this program is licensed as part of the Enterprise
+  versions of CFEngine, the applicable Commercial Open Source License
+  (COSL) may apply to this file if you as a licensee so wish it. See
+  included file COSL.txt.
+*/
+
+#ifndef CFENGINE_PROC_MANAGER_H
+#define CFENGINE_PROC_MANAGER_H
+
+#include <sys/types.h>
+
+/**
+ * Struct to hold information about a subprocess.
+ */
+struct _Subprocess {
+    char *id;             /** unique ID of the subprocess                  [always not %NULL] */
+    char *cmd;            /** command used to create the subprocess        [can be %NULL] */
+    char *description;    /** human-friendly description of the subprocess [can be %NULL] */
+    pid_t pid;            /** PID of the subprocess                        [always >= 0] */
+    int in_fd;            /** FD of the input to the subprocess            [-1 if N/A] */
+    int out_fd;           /** FD of the output from the subprocess         [-1 if N/A] */
+    char lookup_fd;       /** which FD to use for lookup by FD             ['i', 'o' or '\0'] */
```

Comment:
perhaps a simple enum to give a bit more context to the options?
```c
enum LookupByFD {
    IN,
    OUT,
    NONE
};

Reply:
That was my original idea too, but `enum` is an `int` in C, i.e. at least 4 bytes big. `char` is just one byte.

Reply:
If we were using C++ (C++ 11 or later) we could:

```C++
enum LookupByFD: uint8_t
{
    IN,
    OUT,
    NONE
}
```

Without it, the "workaround" is:

```C++
enum _LookupByFD
{
    IN,
    OUT,
    NONE
}

typedef LookupByFD uint8_t;
```

But I agree with your choice, this is not much "nicer".